### PR TITLE
Enforce nextcloud code style | Adding eslint linting

### PR DESCRIPTION
**Pull request by susnux**
Now that our eslint config supports TypeScript we would make use of it, what do you think?

I pushed the fixes for problems found as different commits to make reviewing the changes easier (you could review a single commit a time and compare with the description from the commit message), as the changes are quite huge.

### CODE REVIEW

**Review by skjnldsv — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -41,12 +43,14 @@
...
   "devDependencies": {
     "@nextcloud/browserslist-config": "^2.3.0",
+    "@nextcloud/eslint-config": "^8.2.0",
     "@nextcloud/typings": "^1.6.0",
     "@rollup/plugin-typescript": "^11.0.0",
     "@types/jest": "^29.2.5",
     "@types/node-gettext": "^3.0",
     "@zamiell/typedoc-plugin-not-exported": "^0.2.0",
     "check-es-compat": "^2.2.0",
+    "eslint": "^8.31.0",
```
Comment (skjnldsv) [inline package.json:53]: Not needed as included in `@nextcloud/eslint-config`

**Review by skjnldsv — CHANGES_REQUESTED** [review]
See comment

File: .eslintrc.json [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+				"import/extensions": "off"
+			}
+		},
+		{
+			"files": ["*.config.*js"],
+			"rules": {
+				"n/no-unpublished-import": "off"
+			}
+		}
+	]
```
Comment (skjnldsv) [inline .eslintrc.json:23]: Would be amazing to upstream that to the esling config :rocket:

Comment (skjnldsv) [reply .eslintrc.json:23]: `"cypress/**/*.js",  "cypress/**/*.ts", "*.cy.js", "*.cy.ts"` too (cypress)
`"*.config.*js", "*.config.*ts"`

File: .eslintrc.json [hunk 3]
```diff
@@ -0,0 +1,24 @@
+{
+	"extends": [
+		"@nextcloud"
+	],
+	"ignorePatterns": ["node_modules", "dist"],
```
Comment (skjnldsv) [inline .eslintrc.json:5]: node_modules is already ignored by default alongside the git folder iirc

File: package.json (hunk 1 above)
Comment (susnux) [reply package.json]: :+1:

**Review by skjnldsv — APPROVED** [review]

**Pull request merged**